### PR TITLE
Secure backend socket.io from other applications that can access localhost i.e. browser

### PR DESCRIPTION
**Pull request by Kacper-RF**
### Pull Request Checklist

- [x] I have linked this PR to related GitHub issue.
- [ ] I have updated the CHANGELOG.md file with relevant changes (the file is located at the root of monorepo).

### (Optional) Mobile checklist

Please ensure you completed the following checks if you did any changes to the mobile package:

- [ ] I have run e2e tests for mobile
- [ ] I have updated base screenshots for visual regression tests

### CODE REVIEW

File: packages/desktop/src/main/main.ts [hunk 1]
```diff
@@ -30,6 +30,8 @@ const updaterInterval = 15 * 60_000
 export const isDev = process.env.NODE_ENV === 'development'
 export const isE2Etest = process.env.E2E_TEST === 'true'
 
+const SOCKET_IO_TOKEN = generateJWT()
```
Comment (leblowl) [inline packages/desktop/src/main/main.ts:33]: Why not just use a random number as a token with basic auth? What benefit does a JWT provide here?

**Review by siepra — APPROVED** [review]
Ideally those utility functions for eg. generating random string should be moved to some external files/classes on mobile (obj-c code is already complex enough to be hard to read).
It will work though

File: packages/mobile/ios/Utils.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    for _ in 0..<length {
+        let rand = Int(arc4random_uniform(UInt32(letters.count)))
+        let character = letters[letters.index(letters.startIndex, offsetBy: rand)]
+        randomString.append(character)
+    }
+
+    return randomString
+}
+}
```
Comment (leblowl) [inline packages/mobile/ios/Utils.swift:17]: Same comment as above

> We should probably use something like: https://developer.apple.com/documentation/security/1399291-secrandomcopybytes
>
> and maybe call it `generateSecret` to match other implementations.

Also it looks like this is a duplicate of the same function in AppDelegate.m

**Review by leblowl — CHANGES_REQUESTED** [review]
Nice job getting this all working! I have a few suggestions, but looks good overall.

File: packages/backend/src/utils.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  socketIOSecret: string
+}
+
+// concept
+export const validateOptions = (_options: commander.OptionValues) => {
+  const options = _options as Options
+  if (!options.socketIOSecret) {
+    throw new Error('socketIOSecret is missing in options')
+  }
+}
```
Comment (leblowl) [inline packages/backend/src/utils.ts:23]: I would probably call this file config.ts or options.ts, might make it easier to find

**Review by leblowl — APPROVED** [review]
Looks good, thanks!

**Pull request merged**